### ISSUE TITLE
Test failure: System.Net.Security.Tests.ServerAsyncAuthenticateTest.ServerAsyncAuthenticate_SniSetVersion_Success(version: Tls11)

### ISSUE THREAD

**Issue by v-haren** [dotnet/runtime#41009]
failed in job: [runtime 20200818.76 ](https://dev.azure.com/dnceng/public/_build/results?buildId=777063&view=ms.vss-test-web.build-test-results-tab&runId=24303606&resultId=155678&paneView=debug)

Error message
~~~
System.AggregateException : One or more errors occurred. (One or more errors occurred. (Authentication failed, see inner exception.)) (One or more errors occurred. (Authentication failed, see inner exception.))
---- System.AggregateException : One or more errors occurred. (Authentication failed, see inner exception.)
-------- System.Security.Authentication.AuthenticationException : Authentication failed, see inner exception.
------------ Interop+OpenSsl+SslException : SSL Handshake failed with OpenSSL error - SSL_ERROR_SSL.
---------------- Interop+Crypto+OpenSslCryptographicException : error:1409442E:SSL routines:ssl3_read_bytes:tlsv1 alert protocol version
---- System.AggregateException : One or more errors occurred. (Authentication failed, see inner exception.)
-------- System.Security.Authentication.AuthenticationException : Authentication failed, see inner exception.
------------ Interop+OpenSsl+SslException : SSL Handshake failed with OpenSSL error - SSL_ERROR_SSL.
---------------- Interop+Crypto+OpenSslCryptographicException : error:14209102:SSL routines:tls_early_post_process_client_hello:unsupported protocol


Stack trace
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks) in /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 122
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks, Int32 millisecondsTimeout) in /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 84
   at System.Net.Security.Tests.ServerAsyncAuthenticateTest.ServerAsyncAuthenticate_SniSetVersion_Success(SslProtocols version) in /_/src/libraries/System.Net.Security/tests/FunctionalTests/ServerAsyncAuthenticateTest.cs:line 123
--- End of stack trace from previous location ---
----- Inner Stack Trace #1 (System.AggregateException) -----

----- Inner Stack Trace -----
   at System.Net.Security.SslStream.<ForceAuthenticationAsync>d__171`1[[System.Net.Security.AsyncReadWriteAdapter, System.Net.Security, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a]].MoveNext() in /_/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs:line 393
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks) in /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 96
----- Inner Stack Trace -----
   at Interop.OpenSsl.DoSslHandshake(SafeSslHandle context, ReadOnlySpan`1 input, Byte[]& sendBuf, Int32& sendCount) in /_/src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs:line 290
   at System.Net.Security.SslStreamPal.HandshakeInternal(SafeFreeCredentials credential, SafeDeleteSslContext& context, ReadOnlySpan`1 inputBuffer, Byte[]& outputBuffer, SslAuthenticationOptions sslAuthenticationOptions) in /_/src/libraries/System.Net.Security/src/System/Net/Security/SslStreamPal.Unix.cs:line 114
----- Inner Stack Trace -----

----- Inner Stack Trace #2 (System.AggregateException) -----

----- Inner Stack Trace -----
   at System.Net.Security.SslStream.<ForceAuthenticationAsync>d__171`1[[System.Net.Security.AsyncReadWriteAdapter, System.Net.Security, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a]].MoveNext() in /_/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs:line 393
----- Inner Stack Trace -----
   at Interop.OpenSsl.DoSslHandshake(SafeSslHandle context, ReadOnlySpan`1 input, Byte[]& sendBuf, Int32& sendCount) in /_/src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs:line 290
   at System.Net.Security.SslStreamPal.HandshakeInternal(SafeFreeCredentials credential, SafeDeleteSslContext& context, ReadOnlySpan`1 inputBuffer, Byte[]& outputBuffer, SslAuthenticationOptions sslAuthenticationOptions) in /_/src/libraries/System.Net.Security/src/System/Net/Security/SslStreamPal.Unix.cs:line 114
----- Inner Stack Trace -----
~~~

**Comment by danmosemsft** [comment]
@v-haren can you please include configuration information in issues about test failures? For example this one is `net5.0-Linux-Release-x64-Mono_release-(Debian.10.Amd64.Open)ubuntu.1604.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-10-helix-amd64-bfcd90a-20200121150006`

@wfurt looks like https://github.com/dotnet/runtime/pull/39482

**Comment by v-haren** [comment]
Got it. will be added in the later questions

**Comment by wfurt** [comment]
I will take a look. The PR was clean when submitted but I would not be surprised. 

also note that the links do expire pretty quickly so adding copy of some basic info is generally always useful.

**Comment by karelz** [comment]
Reopening to wait on backport in #41372

**Comment by karelz** [comment]
Fixed in 5.0 by PR #41372
